clang++ on. ubuntu:devel is broken

From https://github.com/votca/votca/actions/runs/23205348911/job/67482253683#step:6:40:
```
-- The CXX compiler identification is Clang 21.1.8
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: /usr/bin/clang++
CMake Error at /usr/share/cmake-4.2/Modules/CMakeTestCXXCompiler.cmake:73 (message):
-- Check for working CXX compiler: /usr/bin/clang++ - broken
  The C++ compiler

    "/usr/bin/clang++"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/__w/votca/votca/builddir/CMakeFiles/CMakeScratch/TryCompile-2qi2C4'
    
    Run Build Command(s): /usr/bin/ninja -v cmTC_74085
    [1/2] /usr/bin/clang++    -MD -MT CMakeFiles/cmTC_74085.dir/testCXXCompiler.cxx.o -MF CMakeFiles/cmTC_74085.dir/testCXXCompiler.cxx.o.d -o CMakeFiles/cmTC_74085.dir/testCXXCompiler.cxx.o -c /__w/votca/votca/builddir/CMakeFiles/CMakeScratch/TryCompile-2qi2C4/testCXXCompiler.cxx
    [2/2] : && /usr/bin/clang++   CMakeFiles/cmTC_74085.dir/testCXXCompiler.cxx.o -o cmTC_74085   && :
    FAILED: [code=1] cmTC_74085 
    : && /usr/bin/clang++   CMakeFiles/cmTC_74085.dir/testCXXCompiler.cxx.o -o cmTC_74085   && :
    /usr/bin/x86_64-linux-gnu-ld.bfd: cannot find -lstdc++: No such file or directory
    clang++: error: linker command failed with exit code 1 (use -v to see invocation)
    ninja: build stopped: subcommand failed.
```

`libstdc++` was installed in https://github.com/votca/buildenv/pull/236, but didn't help.

--- a/ubuntu
+++ b/ubuntu
@@ -9,8 +9,9 @@ ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update && \
     apt-get upgrade -y && \
     apt-get install -y \
-       make cmake valgrind git g++ libexpat-dev libfftw3-dev libboost-all-dev txt2tags ccache gnuplot python3-numpy doxygen vim clang llvm python3-pip python3-lxml \
-       wget libhdf5-dev graphviz pkg-config psmisc libeigen3-dev libxc-dev sudo curl clang-tidy ninja-build libclang-dev llvm-dev libomp-dev libstdc++-13-dev \
+       make cmake valgrind git g++ libexpat-dev libfftw3-dev libboost-program-options-dev libboost-filesystem-dev libboost-regex-dev libboost-timer-dev libboost-test-dev \
+       ccache python3-numpy doxygen vim clang llvm python3-pip python3-lxml \
+       wget libhdf5-dev graphviz pkg-config libeigen3-dev libxc-dev sudo curl clang-tidy ninja-build libclang-dev llvm-dev libomp-dev \
        clang-format software-properties-common zstd libint2-dev libecpint-dev python3-rdkit python3-h5py python3-pytest pybind11-dev python3-xmltodict ase && \
     apt-get purge --autoremove -y && \
     apt-get clean

